pkg/eventsapi/event_container.go

```diff
@@ -0,0 +1,30 @@
+package eventsapi
+
+import (
+	"encoding/json"
+)
+
+type EventContainer struct {
+	EventVersion EventVersion
+	EventData    json.RawMessage
+}
+
+func (ec *EventContainer) UnmarshalEvent() (Event, error) {
+	jsonEventData, err := ec.EventData.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	switch ec.EventVersion {
+	case EventVersion1:
+		var v1Event EventV1
+		err = json.Unmarshal(jsonEventData, &v1Event)
+		return &v1Event, err
+	case EventVersion2:
+		var v2Event EventV2
+		_ = json.Unmarshal(jsonEventData, &v2Event)
```

Comment:
This is probably the error we're actually interested in returning.

Reply:
Good eyes! 620b735